### PR TITLE
Feature/14705 day selection screen part two

### LINKED ISSUE #14705: Android: Blogging Reminders: select days screen
Add a screen where bloggers can set/edit blogging reminders (see screenshot below: it's iOS based for now).

<p align=center>

</p>

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #14705 

This PR adds the remaining view to the day selection screen. In addition it fixes a few UI bugs and adds a few improvements.

- The bottom button is now only disabled when the screen is entered after a new post is created and the selection is empty. In this case the bottom button says "Notify me". If the user enters the day selection from the site settings, the button is always enabled and the button says "Update".
- The days are redrawn based on a diff which makes the UI smoother.
- The primary button is moved outside of the recycler view and always stays visible.
- The label in site settings is fixed

To test:
- Clear app data and log in
- Make sure the flag is turned on
- Create and publish a post
- Notice the Prologue is visible
- Click on Continue
- Notice the day selection is visible
- Notice the primary button is disabled
- Rotate your screen
- Notice the primary button is still visible
- Select a few days
- Notice the primary button is now enabled
- Notice a label appears bellow the day buttons
- Deselect the buttons 
- The label disappears
- Select the buttons again
- Confirm the selection
- Go to site settings
- Notice the "Blogging goals" item has the right amount of days selected

## Regression Notes
1. Potential unintended areas of impact
- None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by osullivanchris — CHANGES_REQUESTED** [review]
Looking pretty great @planarvoid . Checked light mode/dark mode, landscape/portrait on phone.

Couple small details
- The grey colour used for the background of unselected days, and the tip, looks darker in the build than in the mocks. Can we make it lighter?
- I find the text in the tip a bit too small/weak looking. I think you have followed the design spec which says caption (12dp) but I reckon it might look better at body2 (14dp). Do-able?
- Know you had trouble with the shadowed fixed button. I would be ok with a divider. I also think the shadow is only needed on scroll. Personally I wouldn't display it immediately...if possible.

Can't spot anything else right now. The day updating (N times / week) as I make selections is really slick, works very well.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingReminderBottomSheetFragment.kt [hunk 1]
```diff
@@ -8,43 +8,65 @@ import android.view.ViewGroup
...
+                    override fun onStateChanged(view: View, newState: Int) {
+                        if (newState == BottomSheetBehavior.STATE_COLLAPSED) {
+                            behavior.state = BottomSheetBehavior.STATE_HIDDEN
+                        }
+                    }
+
+                    override fun onSlide(view: View, v: Float) {
+                        // Not needed
+                    }
+                })
```
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingReminderBottomSheetFragment.kt]: Unless I'm missing something, I think we can probably use `behavior.skipCollapsed = true` here instead.

**Review by renanferrari — COMMENTED** [review]
Thanks for the changes! The app is working as expected and the code is looking good overall. I just left one minor comment below.

**Comment by renanferrari** [comment]
@planarvoid Just a heads up that I have merged `develop` into this so I could properly review #14768. I tried doing that in a separate, intermediary branch, but it didn't work. Hope you don't mind!

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingReminderBottomSheetFragment.kt (hunk 1 above)
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingReminderBottomSheetFragment.kt]: good tip 👍  fixed in bf09aed

**Comment by planarvoid** [comment]
Thanks for the review @osullivanchris ! 
I've fixed all the issues you've mentioned so it's now ready for another check: 

Screenshot_1624355837
Screenshot_1624355784

**Comment by planarvoid** [comment]
thanks for the review @renanferrari ! I've implemented your suggestion 👍

**Comment by osullivanchris** [comment]
Thanks for sharing the update @planarvoid 

checked on light/dark, portrait/landscape

- bigger details text on the 'tip' looks good 👍 
- good job getting the shadow to display only on scroll, hope it didn't drive you crazy 👍 
- background colour on unselected days now looks correct 👍 
- the background colour for the tip box still looks too dark. Could you make it the same colour as the unselected days? 🙏

**Review by renanferrari — APPROVED** [review]
LGTM 👍

**Comment by planarvoid** [comment]
Thanks for the review @osullivanchris . It should all be fixed now 👍  

Screenshot 2021-06-23 at 10 14 33

**Comment by osullivanchris** [comment]
Thanks @planarvoid  all good for me here!

**Review by osullivanchris — APPROVED** [review]

**Pull request merged**